### PR TITLE
Holybor durandal rev 1 has bmi088->icm20602

### PR DESCRIPTION
**Pull request by davids5**
@bkueng, @dagar  Would you please check the rotations? 

Unless there is an error in the datasheet (or my interpretation of it) it appears the Y
orientation of the icm20602 is opposite of the icm20689 .

I found that the icm20602 has 
image

The icm20689 has
image

But both have the same transform applied.
icm20602:
https://github.com/PX4/PX4-Autopilot/blob/master/src/drivers/imu/invensense/icm20602/ICM20602.cpp#L637-L642

icm20689:
https://github.com/PX4/PX4-Autopilot/blob/master/src/drivers/imu/invensense/icm20689/ICM20689.cpp#L580-L585

This  is the layout:

image

### CODE REVIEW

File: boards/holybro/durandal-v1/src/spi.cpp [hunk 1]
```diff
@@ -35,28 +35,52 @@
...
-	}, {GPIO::PortE, GPIO::Pin3}),
-	initSPIBus(SPI::Bus::SPI2, {
-		initSPIDevice(SPIDEV_FLASH(0), SPI::CS{GPIO::PortF, GPIO::Pin5})
+constexpr px4_spi_bus_all_hw_t px4_spi_buses_all_hw[BOARD_NUM_SPI_CFG_HW_VERSIONS] = {
+	initSPIHWVersion(HW_VER_REV(0, 0), {
+		initSPIBus(SPI::Bus::SPI1, {
+			initSPIDevice(DRV_IMU_DEVTYPE_ICM20689, SPI::CS{GPIO::PortF, GPIO::Pin2}, SPI::DRDY{GPIO::PortB, GPIO::Pin4}),
+			initSPIDevice(DRV_GYR_DEVTYPE_BMI088, SPI::CS{GPIO::PortF, GPIO::Pin4}, SPI::DRDY{GPIO::PortB, GPIO::Pin14}),
+			initSPIDevice(DRV_ACC_DEVTYPE_BMI088, SPI::CS{GPIO::PortG, GPIO::Pin10}, SPI::DRDY{GPIO::PortB, GPIO::Pin15}),
+			initSPIDevice(DRV_DEVTYPE_UNUSED, SPI::CS{GPIO::PortH, GPIO::Pin5}),
```
Comment (dagar) [inline boards/holybro/durandal-v1/src/spi.cpp:44]: What is PH5?

**Comment by vincentpoont2** [comment]
Tested the FW [Here](from https://github.com/PX4/PX4-Autopilot/suites/4329238218/artifacts/113389364) with a REV1 Durandal with ICM20602. Unable to recognize the ICM20602.

[PX4 PR Test Console Output.txt](https://github.com/PX4/PX4-Autopilot/files/7560708/PX4.PR.Test.Console.Output.txt)
le

**Comment by davids5** [comment]
@vincentpoont2 - no my bad....fix coming

**Comment by davids5** [comment]
@vincentpoont2 - Please retest

**Comment by vincentpoont2** [comment]
Hi @davids5, tested and both REV0 & REV1 Hardware can recognize the IMUs, Attached logs for moving the FC on the desk. Will test flight soon.

Durandal_test2
[Durandal PX4 FW Test Log.zip](https://github.com/PX4/PX4-Autopilot/files/7578803/Durandal.PX4.FW.Test.Log.zip)

**Comment by davids5** [comment]
> Hi @davids5, tested and both REV0 & REV1 Hardware can recognize the IMUs, Attached logs for moving the FC on the desk. Will test flight soon.
> 
> Durandal_test2 [Durandal PX4 FW Test Log.zip](https://github.com/PX4/PX4-Autopilot/files/7578803/Durandal.PX4.FW.Test.Log.zip)

@vincentpoont2 - I can not open the ulog files. Plotjuggel has an exception and review.px4.io makes the files as bad.

Please upload them to https://review.px4.io/

You can also attach the console 

and issue `listener sensor_accel` in 6 pos

**Comment by vincentpoont2** [comment]
Tested on REV0 & REV1 on Console. Everything looks good.
[Duranndal REV1 Test.txt](https://github.com/PX4/PX4-Autopilot/files/7587666/Duranndal.REV1.Test.txt)

**Comment by davids5** [comment]
@vincentpoont2 

It is hard to understand what the orientation of the FMU is:

Instance 0:
 sensor_accel
    timestamp: 127400838 (0.001086 seconds ago)
    timestamp_sample: 127400523 (315 us before timestamp)
    device_id: 3932170 (Type: 0x3C, SPI:1 (0x00))
    `x: -0.0253`
    y: 0.3356
    z: 9.5189
    temperature: 30.8262
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5


Instance 1:
 sensor_accel
    timestamp: 127414079 (0.000095 seconds ago)
    timestamp_sample: 127413860 (219 us before timestamp)
    device_id: 3670026 (Type: 0x38, SPI:1 (0x00))
    `x: 0.0742`
    y: 0.0976
    z: 9.7214
    temperature: 30.6842
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5


We need to see X (nose) Down, Y (right side) Down, and Z (top) Down

We should see ~9.8 on X Y and Z.  The signs should match from Instance 0 to Instance 1

**Comment by vincentpoont2** [comment]
Sorry @davids5 , I forgot to mention.
First listener sensor_accel command was with FC right side up. 
Second listener sensor_accel command was with the FC upside down.

**Comment by vincentpoont2** [comment]
Should I test the all other orientations?

**Comment by davids5** [comment]
> Should I test the all other orientations?

Yes we need to see X (nose) Down, Y (right side) Down, and Z (top) Down

**Comment by vincentpoont2** [comment]
Hi @davids5 , attached is the console output for the REV1 Durandal.
[Durandal REV1 Test PX4 Console -1.2.txt](https://github.com/PX4/PX4-Autopilot/files/7597763/Durandal.REV1.Test.PX4.Console.-1.2.txt)

It was reported that using this firmware, these issue were encountered.
-　The flight mode cannot be switched after unlocking, and the motor shakes severely. ESC was Calibrated but didn't improve.
-　Saving the flight mode settings will cause QGC ground station to closed
-　Flight Log can't open.
-   After the flight controller is unlock , it cannot be lock, battery has be to unplug.
QGC 4.1.4 stable.

I am trying to verify these issue at the moment. 

Can we send you one of the REV1 Durandal Flight Controller?

**Comment by davids5** [comment]
> 

@vincentpoont2 - yes. Can you work on getting logs and we will ask @dagar to take a look. 

The concern I has about the rotation is must be an error in the data sheet.

The value on the  ICM20689 (0x3C) Seem to be to be high @dagar any ideas here? 

```
NuttShell (NSH) NuttX-10.1.0
nsh> work_queue status

Work Queue: 10 threads                          RATE        INTERVAL
|__ 1) wq:rate_ctrl    
|   |__ 1) mc_rate_control                  796.6 Hz         1255 us
|   |__ 2) px4io                            357.3 Hz         2799 us
|   \__ 3) vehicle_angular_velocity         796.9 Hz         1255 us
|__ 2) wq:SPI1         
|   |__ 1) icm20602                         799.0 Hz         1252 us
|   \__ 2) icm20689                         793.7 Hz         1260 us
|__ 3) wq:SPI4         
|   \__ 1) ms5611                            99.8 Hz        10024 us
|__ 4) wq:I2C3         
|   \__ 1) ist8310                           46.7 Hz        21423 us



nsh> listener sensor_accel

#define DRV_IMU_DEVTYPE_ICM20602 0x38
#define DRV_IMU_DEVTYPE_ICM20689 0x3C

TOPIC: sensor_accel 2 instances

Instance 0:
 sensor_accel
    timestamp: 38229194 (0.000712 seconds ago)
    timestamp_sample: 38228931 (263 us before timestamp)
    device_id: 3932170 (Type: 0x3C, SPI:1 (0x00))
    x: -0.2705
    y: 0.2178
    z: -10.4296
    temperature: 28.0845
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5


Instance 1:
 sensor_accel
    timestamp: 38241925 (0.000189 seconds ago)
    timestamp_sample: 38241555 (370 us before timestamp)
    device_id: 3670026 (Type: 0x38, SPI:1 (0x00))
    x: -0.2236
    y: 0.1096
    z: -9.9364
    temperature: 28.1334
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5

nsh> listener sensor_accel

TOPIC: sensor_accel 2 instances

Instance 0:
 sensor_accel
    timestamp: 54044173 (0.000883 seconds ago)
    timestamp_sample: 54043921 (252 us before timestamp)
    device_id: 3932170 (Type: 0x3C, SPI:1 (0x00))
    x: 0.1211
    y: 0.3481
    z: 9.5189
    temperature: 27.9865
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5


Instance 1:
 sensor_accel
    timestamp: 54056422 (0.000692 seconds ago)
    timestamp_sample: 54056204 (218 us before timestamp)
    device_id: 3670026 (Type: 0x38, SPI:1 (0x00))
    x: 0.1015
    y: 0.0469
    z: 9.7406
    temperature: 28.1824
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5

nsh> listener sensor_accel

TOPIC: sensor_accel 2 instances

Instance 0:
 sensor_accel
    timestamp: 59666839 (0.003088 seconds ago)
    timestamp_sample: 59666586 (253 us before timestamp)
    device_id: 3932170 (Type: 0x3C, SPI:1 (0x00))
    x: -9.8742
    y: 0.2394
    z: -0.2092
    temperature: 27.9865
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5


Instance 1:
 sensor_accel
    timestamp: 59681494 (0.000668 seconds ago)
    timestamp_sample: 59681275 (219 us before timestamp)
    device_id: 3670026 (Type: 0x38, SPI:1 (0x00))
    x: -9.8272
    y: 0.0775
    z: 0.2777
    temperature: 28.1775
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5

nsh> listener sensor_accel

TOPIC: sensor_accel 2 instances

Instance 0:
 sensor_accel
    timestamp: 66293737 (0.002218 seconds ago)
    timestamp_sample: 66293482 (255 us before timestamp)
    device_id: 3932170 (Type: 0x3C, SPI:1 (0x00))
    x: 9.7760
    y: 0.3509
    z: -0.4280
    temperature: 28.2313
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5


Instance 1:
 sensor_accel
    timestamp: 66307164 (0.001132 seconds ago)
    timestamp_sample: 66306945 (219 us before timestamp)
    device_id: 3670026 (Type: 0x38, SPI:1 (0x00))
    x: 9.7578
    y: 0.0392
    z: -0.1422
    temperature: 28.1971
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5

nsh> listener sensor_accel

TOPIC: sensor_accel 2 instances

Instance 0:
 sensor_accel
    timestamp: 71053494 (0.001665 seconds ago)
    timestamp_sample: 71053245 (249 us before timestamp)
    device_id: 3932170 (Type: 0x3C, SPI:1 (0x00))
    x: -0.1455
    y: -9.5644
    z: -0.3572
    temperature: 28.1334
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5


Instance 1:
 sensor_accel
    timestamp: 71066554 (0.000622 seconds ago)
    timestamp_sample: 71066338 (216 us before timestamp)
    device_id: 3670026 (Type: 0x38, SPI:1 (0x00))
    x: -0.1245
    y: -9.7669
    z: -0.0354
    temperature: 28.2020
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5

nsh> listener sensor_accel

TOPIC: sensor_accel 2 instances

Instance 0:
 sensor_accel
    timestamp: 78598325 (0.002647 seconds ago)
    timestamp_sample: 78598094 (231 us before timestamp)
    device_id: 3932170 (Type: 0x3C, SPI:1 (0x00))
    x: -0.0727
    y: 10.0604
    z: -0.3768
    temperature: 28.3293
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5


Instance 1:
 sensor_accel
    timestamp: 78612892 (0.000928 seconds ago)
    timestamp_sample: 78612673 (219 us before timestamp)
    device_id: 3670026 (Type: 0x38, SPI:1 (0x00))
    x: -0.0775
    y: 9.7818
    z: -0.0057
    temperature: 28.3341
    error_count: 0
    clip_counter: [0, 0, 0]
    samples: 5
```

**Comment by davids5** [comment]
@dagar - rebased on master.

**Comment by vincentpoont2** [comment]
@davids5 We shipped you a REV1 Durandal. 

We still cant get the log to read properly. https://review.px4.io/upload would say its 
> HTTP Error 400: Failed to parse the file. It is most likely corrupt.

Attached is the console output. 
[Durandal REV1 Test PX4 Console - 20211125.txt](https://github.com/PX4/PX4-Autopilot/files/7604570/Durandal.REV1.Test.PX4.Console.-.20211125.txt)

**Comment by davids5** [comment]
> @davids5 We shipped you a REV1 Durandal.
> 
> We still cant get the log to read properly. https://review.px4.io/upload would say its
> 
> > HTTP Error 400: Failed to parse the file. It is most likely corrupt.
> 
> Attached is the console output. [Durandal REV1 Test PX4 Console - 20211125.txt](https://github.com/PX4/PX4-Autopilot/files/7604570/Durandal.REV1.Test.PX4.Console.-.20211125.txt)

I tested master on the Rev 0 HW yesterday and it loaded fine
https://review.px4.io/plot_app?log=c00d583b-65af-460d-a55e-385fc1a80b38 

I will take a look on the new HW when it arrives.

**Review by dagar — APPROVED** [review]

**Pull request merged**